projectq/cengines/_twodmapper.py

```diff
@@ -87,11 +101,32 @@ def __init__(self, num_rows, num_columns, storage=1000,
             num_optimization_steps(int): Number of different permutations to
                                          of the matching to try and minimize
                                          the cost.
+        Raises:
+            RuntimeError: if incorrect `mapped_ids_to_backend_ids` parameter
         """
         BasicMapperEngine.__init__(self)
         self.num_rows = num_rows
         self.num_columns = num_columns
         self.num_qubits = num_rows * num_columns
+        # Internally we use the mapped ids until sending a command.
+        # Before sending we use this map to translate to backend ids:
+        self._mapped_ids_to_backend_ids = mapped_ids_to_backend_ids
+        if self._mapped_ids_to_backend_ids is None:
+            self._mapped_ids_to_backend_ids = dict()
+            for i in range(self.num_qubits):
+                self._mapped_ids_to_backend_ids[i] = i
+        if (not (set(self._mapped_ids_to_backend_ids.keys()) ==
+                 set(list(range(self.num_qubits)))) or not (
+                 len(set(self._mapped_ids_to_backend_ids.values())) ==
+                 self.num_qubits)):
+            raise RuntimeError("Incorrect mapped_ids_to_backend_ids parameter")
+        self._backend_ids_to_mapped_ids = dict()
+        for mapped_id, backend_id in self._mapped_ids_to_backend_ids.items():
+            self._backend_ids_to_mapped_ids[backend_id] = mapped_id
+        # As we use internally the mapped ids which are in row-major order,
+        # we have an internal current mapping which maps from logical ids to
+        # these mapped ids:
+        self._current_row_major_mapping = deepcopy(self.current_mapping)
```

Comment:
Isn't this just `None`?

Reply:
Just in case someone changes the initial value of BasicEngine otherwise it will be hard to tell if something goes wrong in this mapper.